lib/Backend/GlobOpt.cpp

```diff
@@ -8988,7 +8988,11 @@ GlobOpt::OptConstFoldUnary(
     case Js::OpCode::Ld_A:
         if (instr->HasBailOutInfo())
         {
-            Assert(instr->GetBailOutKind() == IR::BailOutExpectingInteger);
+            //The profile data for switch expr can be string and in GlobOpt we realize it can be a int.
```

Comment:
nit:Sicne, we are constant folding, the a" should be "is an" to convey the right information